src/Sarif.Driver/Sdk/MultithreadedAnalyzeCommandBase.cs

```diff
@@ -859,7 +896,20 @@ private SupportedPlatform GetCurrentRunningOS()
                 return context;
             }
 
+            // If computeHashes exists and we already added in the cache,
+            // we can just return the context.
+            if (_computeHashes && _analysisLoggerCache.ContainsKey(context.Hashes.Sha256))
+            {
+                return context;
+            }
+
             context.Logger.AnalyzingTarget(context);
+
+            if (_computeHashes)
```

Comment:
>_computeHashes [](http://example.com/codeflow?start=16&length=14)

if we have hashes enabled, we will save the logger in the cache which will be re-used in the future